code-orange/MainViewController.swift

```diff
@@ -137,11 +142,21 @@ class MainViewController: UIViewController {
     view.setAutoLayoutHeight(1)
     return view
   }()
+
+  private lazy var drawerView: DrawerView = {
+    let drawerView = DrawerView()
+    drawerView.translatesAutoresizingMaskIntoConstraints = false
+    return drawerView
+  }()
```

Comment:
I have one concern about this: Now that the drawer is not optional, you remove the check if it was nil before... meaning each drawer content can run over the previous one, and we have  no priorities defined.
Lets make sure we do not allow that, and then define a priority of how should be displayed 
[critical]

Reply:
I agree that the drawer content is free to change but I think that's not a problem. Unlike network alerts, for example, in this case the only one triggering the drawer is MainViewController, and so it's up to it to define flow and what dismisses what (which is by def the logic of a VC). If it doesn't manage priorities actively we would just either replace the content of the drawer while it's open or open it with our own content and that also looks and feels ok - try starting the app with this branch then clicking the status button to see that

Reply:
I think it is really easy to fix (or create foundations for making it easier to manage)... so we can do it now.
for example; having an enum indicating what overlay we are presenting now (or none) in order to make this more manageable

Reply:
done 🚀 🚀